src/test/java/org/symphonyoss/integration/webhook/salesforce/parser/v1/V1SalesforceParserFactoryTest.java

```diff
@@ -112,4 +120,28 @@ public void testAccountStatusParser() throws IOException {
     assertEquals(accountStatusParser.getEvents(), factory.getParser(payload).getEvents());
   }
 
+  @Test(expected = SalesforceParseException.class)
+  public void testInvalidPayload() throws IOException {
+    WebHookPayload payload = new WebHookPayload(Collections.<String, String>emptyMap(), Collections.<String, String>emptyMap(), "invalid_payload");
+
+    factory.getParser(payload).getEvents();
+  }
+
+  @Test
+  public void testUnregistredParser() throws IOException{
+    String xml = readFile("executiveReport.xml");
+    WebHookPayload payload = new WebHookPayload(Collections.<String, String>emptyMap(), Collections.<String, String>emptyMap(), xml);
+    WebHookParser webHookParser = factory.getParser(payload);
+
+    Assert.assertEquals(0, factory.getParser(payload).getEvents().size());
```

Comment:
You should change 'factory.getParser(payload).getEvents().size()' to 'webHookParser.getEvents().size()'